packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/utils/find_shared_origin_objects.ts

```diff
@@ -80,17 +90,56 @@ export async function findSharedOriginObjects(
   return objectsMap;
 }
 
-function createAliasKueryFilter(objects: Array<{ type: string; origin: string }>) {
+function createOriginKueryFilter(
+  objects: ObjectOrigin[],
+  purpose?: SavedObjectsCollectMultiNamespaceReferencesPurpose
+) {
   const { buildNode } = esKuery.nodeTypes.function;
   // Note: these nodes include '.attributes' for type-level fields because these are eventually passed to `validateConvertFilterToKueryNode`, which requires it
   const kueryNodes = objects
-    .reduce<unknown[]>((acc, { type, origin }) => {
+    .reduce<unknown[]>((acc, { type, id, origin }) => {
       // Escape Kuery values to prevent parsing errors and unintended behavior (object types/IDs can contain KQL special characters/operators)
-      const match1 = buildNode('is', `${type}.id`, esKuery.escapeKuery(`${type}:${origin}`)); // here we are looking for the raw document `_id` field, which has a `type:` prefix
-      const match2 = buildNode('is', `${type}.originId`, esKuery.escapeKuery(origin)); // here we are looking for the saved object's `originId` field, which does not have a `type:` prefix
-      acc.push([match1, match2]);
+
+      // Look for objects with an ID that matches the origin or ID (has a `type:` prefix)
+      const idMatchesOrigin = buildNode(
+        KQL_FUNCTION_IS,
+        `${type}.id`,
+        esKuery.escapeKuery(`${type}:${origin || id}`)
+      );
+
+      // Look for objects with an `originId` that matches the origin or ID (does not have a `type:` prefix)
+      const originMatch = buildNode(
+        KQL_FUNCTION_IS,
+        `${type}.originId`,
+        esKuery.escapeKuery(origin || id)
+      );
+
+      // If we are updating an object's spaces (as opposed to copying)...
+      if (purpose === 'updateObjectsSpaces') {
+        // we never want to match if the raw document `_id` fields.
```

Comment:
This phrase doesn't make sense to me. Do you mean:
"We never want to match ~~if~~ _on_ the raw document's "_id" field"?